webhook/chaos_handler.go

```diff
@@ -0,0 +1,87 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type ChaosHandlerMutator struct {
+	Client  client.Client
+	Log     *zap.SugaredLogger
+	Image   string
+	Timeout time.Duration
+	decoder *admission.Decoder
+}
+
+func (m *ChaosHandlerMutator) InjectDecoder(d *admission.Decoder) error {
+	m.decoder = d
+
+	return nil
+}
+
+func (m *ChaosHandlerMutator) Handle(ctx context.Context, req admission.Request) admission.Response {
+	pod := &corev1.Pod{}
+
+	// ensure decoder is set
+	if m.decoder == nil {
+		m.Log.Errorw("webhook decoder seems to be nil while it should not, aborting")
+
+		return admission.Errored(http.StatusInternalServerError, nil)
+	}
+
+	// decode pod object
+	err := m.decoder.Decode(req, pod)
+	if err != nil {
+		m.Log.Errorw("error decoding pod object", "error", err, "pod", pod.Name, "namespace", pod.Namespace)
+
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	// define pod name for logs
+	// if the pod is created from a replicaset (deployment, statefulset)
+	// the name won't be populated yet because it relies on the name generation
+	podName := pod.Name
+	if podName == "" {
+		podName = pod.ObjectMeta.GenerateName
```

Comment:
This is only going to be a prefix of the full pod name. That's fine because we're only using it to log, afaict, but is that worth mentioning explicitly in the comments?